testing-common/src/main/groovy/io/opentelemetry/instrumentation/test/base/HttpClientTest.groovy

```diff
@@ -104,18 +129,18 @@ abstract class HttpClientTest extends InstrumentationSpecification {
    * can be used to block on a result, for example:
    *
    * @Override
-   * int doRequest(String method, URI uri, Map<String, String headers = [:]) {
+   * int sendRequest(REQUEST request, String method, URI uri, Map<String, String> headers) {
    *   CompletableFuture<Integer> result = new CompletableFuture<>()
    *   doRequestWithCallback(method, uri, headers) {
    *     result.complete(it.statusCode())
    *   }
    *   return result.join()
    * }
    */
```

Comment:
Can you update the whole Javadoc so that is does not mention `new Request()` or `doRequest()`?

Also, would it make sense to refactor `doRequestWithCallback()` in a similar way to `doRequest()`? Make it final and split into `buildRequest()` + `sendRequestWithCallback()`